### DISCUSSION TITLE
How i can hide some fields from rendered html

### DISCUSSION THREAD

**Discussion by floloz** [useblocks/sphinx-needs#1357] [Q&A]
Hello guys,

I'm Flo, connected from Rennes, France.

I'm trying to push in my company to use sphinx-needs for requirement management and use it as part of our documentations.
I've got a small issue, if anybody can help me.

So i want to use as much as possible the built-in layouts, the only issue i have is that i'm not able to select what fields are visible in the rendered html (or PDF). For instance having the field 'style' visible seems a bit awkward to me.

I went through the doc, i could not find any solution, except creating my own layout, which would be acceptable, but it will probably lead to new issues for me, so i"d like to clarify that first.

Am i missing something?

Many thanks for your help.

Best,

Flo.

**Comment by danwos** [comment]
Hey Flo,

having your own layout is the only solution to hide specific options from the output.

It's also quite easy to extend it, if you start with the default one:
```python
needs_layouts = {
  "my_layout": {
          "grid": "simple",
          "layout": {
              "head": [
                  '<<meta("type_name")>>: **<<meta("title")>>** <<meta_id()>>  <<collapse_button("meta", '
                  'collapsed="icon:arrow-down-circle", visible="icon:arrow-right-circle", initial=False)>> '
              ],
              "meta": ["<<meta_all(no_links=True)>>", "<<meta_links_all()>>"],
          },
      },
  }
```
Source: [default.py](https://github.com/useblocks/sphinx-needs/blob/c5acec616c76e8873520796e1b91c7d2e9e88cbc/sphinx_needs/defaults.py#L39), contains the configs for our default layouts, and is a good place for layout examples.

`<<meta_all(no_links=True)>>` does most of the magic here and adds all options with a value to the output. If you want to avoid certain options, just use the `exclude` parameter: `<<meta_all(no_links=True, exclude='tags', 'status', 'my_custom_option')>>`
Source: [meta_all](https://sphinx-needs.readthedocs.io/en/latest/layout_styles.html#sphinx_needs.layout.LayoutHandler.meta_all)

I guess you can also name the layout `clean`, which should override the default layout.
But I haven't tested this.

Otherwise you can use `needs_global_options` to set the layout automatically to the new one.
```python
needs_global_options = {
   'layout': 'my_layout'
}
```
Source: [needs_global_option config](https://sphinx-needs.readthedocs.io/en/latest/configuration.html#needs-global-options)

**Comment by ubmax** [comment]
Hi Flo,

Welcome to the Sphinx-Needs community, and thank you for bringing Sphinx-Needs to your company! It’s great to hear about your initiative.

We’d love to have a chat to discuss your current challenges and offer support with materials and best practices from other companies already using Sphinx-Needs. Just drop us a message at mail@useblocks.com, and we’ll be happy to connect!

Looking forward to supporting you on your Sphinx-Needs journey.

Best,
Max
